mhs/mhs/common/state/dynamo_persistence_adapter.py

```diff
@@ -0,0 +1,111 @@
+import json
+import traceback
+
+import boto3
+import utilities.integration_adaptors_logger as log
+
+from mhs.common.state.persistence_adapter import PersistenceAdapter
+
+logger = log.IntegrationAdaptorsLogger('DYNAMO_PERSISTENCE')
+
+
+class RecordCreationError(RuntimeError):
+    def __init__(self):
+        pass
+
+
+class RecordDeletionError(RuntimeError):
+    def __init__(self):
+        pass
+
+
+class RecordRetrievalError(RuntimeError):
+    def __init__(self):
+        pass
+
+
+class InvalidTableError(RuntimeError):
+    def __init__(self):
+        pass
+
+
+class DynamoPersistenceAdapter(PersistenceAdapter):
+    """
+    Class responsible for persisting items into DynamoDB.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Constructs a DynamoDB version of a
+        :class:`PersistenceAdapter <mhs.common.state.persistence_adapter.PersistenceAdapter>`.
+        The kwargs provided should contain the following information:
+          * state_table: The name used to refer to the table containing state items.
+          * state_arn: The Amazon Resource Name used to identify the dynamo table containing state items.
+          * sync_async_table: The name used to refer to the table containing sync_async message items.
+          * sync_async_arn: The Amazon Resource Name used to identify the dynamo table containing sync_async message
+            items.
+        :param kwargs: The key word arguments required for this constructor.
+        """
+        state_table = kwargs.get('state_table', "state")
+        state_arn = kwargs.get('state_arn')
+        sync_async_table = kwargs.get('sync_async_table', "sync_async")
+        sync_async_arn = kwargs.get('sync_async_arn')
+
+        dynamo_db = boto3.resource('dynamodb')
+
+        self.tables = {
+            state_table: dynamo_db.Table(state_arn),
+            sync_async_table: dynamo_db.Table(sync_async_arn)
+        }
+
+    def add(self, table_name, key, item):
+        try:
+            response = self._resolve_table(table_name).put_item(
+                Item={"key": key, "data": json.dumps(item)},
+                ReturnValues='ALL_OLD'
+            )
+            if 'Attributes' not in response:
+                logger.info("000", 'No previous record found: {key}', {"key": key})
+                return None
+            return json.loads(response.get('Attributes', {}).get('data'))
+        except Exception as e:
+            logger.error("001", 'Error creating record: {exception}', {"exception": traceback.format_exc()})
+            raise RecordCreationError from e
+
+    def get(self, table_name, key):
+        logger.info("002", 'Getting record for key {key}', {"key": key})
```

Comment:
This will resolve as 'Getting record for key key=qweqweqwe', doesn't matter too much but it might look better as 
`logger.info("000", 'Getting record for {key}')`.